### DISCUSSION TITLE
TimeEqMatcher appears to be not needed (anymore?)

### DISCUSSION THREAD

**Discussion by akostadinov** [rubocop-hq/rubocop-rspec#1997] [Ideas]
I see that TimeEqMatcher is described as:
```
      # Use `eq_time` matcher with timestamps in specs.
      #
      # @example
      #   # bad
      #   expect(user.created_at).to eq(Time.zone.now)
      #
      #   # good
      #   expect(user.created_at).to eq_time(Time.zone.now)
```

I don't see a point in this because Time comparison in Ruby is smart enough to adjust for time zones.

The time zone in the Time object only matters for presentation but in comparisons the adjusted actual time is being used.

```
[12] pry(main)> Time.at(123134253)
=> 1973-11-26 05:57:33 +0200
[13] pry(main)> t1 = Time.at(123134253)
=> 1973-11-26 05:57:33 +0200
[14] pry(main)> t2 = Time.at(123134253)
=> 1973-11-26 05:57:33 +0200
[15] pry(main)> t2.utc
=> 1973-11-26 03:57:33 UTC
[16] pry(main)> t1 == t2
=> true
[17] pry(main)> t1
=> 1973-11-26 05:57:33 +0200
[18] pry(main)> t2
=> 1973-11-26 03:57:33 UTC
```

So I'm wondering why is this rule implemented. Was it needed in the past, or does it still have practical value?

**Comment by Earlopain** [comment]
RuboCop has no such rule, it seems to be a discourse-specific cop https://github.com/discourse/rubocop-discourse. I'm not sure if rspec even has `eq_time` by default.
